extensions/analyticsdx-vscode-core/src/commands/createDashboardLWC.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { LightningComponentOptions, TemplateService, TemplateType } from '@salesforce/templates';
+import { CreateUtil } from '@salesforce/templates/lib/utils';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import { showConfirmModal } from '../util';
+import { getRootWorkspacePath, hasRootWorkspace } from '../util/rootWorkspace';
+import {
+  CancelResponse,
+  CompositeParametersGatherer,
+  ContinueResponse,
+  FixedValueGatherer,
+  LibraryCommandletExecutor,
+  ParametersGatherer,
+  PostconditionChecker,
+  SfdxCommandlet,
+  sfdxOutputChannel,
+  sfdxWorkspaceChecker
+} from './commands';
+import { OutputDirGatherer, OutputDirType } from './gatherers/outputDirGatherer';
+
+type LWCTemplateType = 'analyticsDashboard' | 'analyticsDashboardWithStep';
+
+type LWCOptions = {
+  fileName: string;
+  /** Relative workspace path to save the component */
+  outputDir: string;
+  template: LWCTemplateType;
+};
+
+class CreateDashboardLWCExecutor extends LibraryCommandletExecutor<LWCOptions> {
+  constructor() {
+    super(nls.localize('create_dashboard_lwc_execution_name'), 'analytics_dashboard_lwc_create', sfdxOutputChannel());
+    this.showChannelOutput = false;
+  }
+
+  public async run(response: ContinueResponse<LWCOptions>): Promise<boolean> {
+    try {
+      const templateOptions: LightningComponentOptions = {
+        outputdir: response.data.outputDir,
+        componentname: response.data.fileName,
+        template: response.data.template,
+        type: 'lwc',
+        internal: false
+      };
+      const result = await TemplateService.getInstance(getRootWorkspacePath()).create(
```

Comment:
This uses the same node module that `sfdx force:lightning:component:create` uses